package.json

```diff
@@ -197,6 +197,11 @@
                 "when": "resourceScheme == 'vscode-interactive'",
                 "command": "interactive.execute"
             },
+            {
+                "key": "escape",
+                "when": "resourceScheme == 'vscode-interactive' && !suggestWidgetVisible",
```

Comment:
We also need to include `!editorHoverVisible`, `!exceptionWidgetVisible`, `!editorHasMultipleSelections`, `!inlineSuggestionVisible`, `!isComposing`, `!inSnippetMode`, `!parameterHintsVisible`, `!renameInputVisible`. (I went through default keybindings for Escape.)

Reply:
Thanks for pointing that out, I went through the default keybindings for escape and came up with this list. IIRC there was a problem with excessively complex when clauses resulting in perf problems, and I'm not sure all of these are strictly required:

resourceScheme == 'vscode-interactive' &&
 !editorHoverVisible &&
 !suggestWidgetVisible &&
 !isComposing &&
 !inSnippetMode &&
 !exceptionWidgetVisible &&
 !selectionAnchorSet &&
 !LinkedEditingInputVisible &&
 !renameInputVisible &&
 !editorHasSelection &&
 !accessibilityHelpWidgetVisible &&
 !breakpointWidgetVisible &&
 !findWidgetVisible &&
 !markersNavigationVisible &&
 !parameterHintsVisible &&
 !editorHasMultipleSelections &&
 !notificationToastsVisible